rust/datafusion/src/physical_plan/hash_aggregate.rs

```diff
@@ -48,9 +48,12 @@ use fnv::FnvHashMap;
 /// Hash aggregate modes
 #[derive(Debug, Copy, Clone)]
 pub enum AggregateMode {
+    /// Aggregate mode without any partial/intermediate aggregation
+    NoPartial,
```

Comment:
A naming idea: `SinglePartition`

Reply:
Spark uses the term `Complete`. Our existing names `Partial` and `Final` were based on Spark's naming.

Reply:
```
/**
 * An [[AggregateFunction]] with [[Complete]] mode is used to evaluate this function directly
 * from original input rows without any partial aggregation.
 * This function updates the given aggregation buffer with the original input of this
 * function. When it has processed all input rows, the final result of this function is returned.
 */
case object Complete extends AggregateMode
```

Reply:
Both suggestions sound good. Let me know if there's a preference one way or the other. For now I will go ahead with `Complete` for consistency with Spark.